### PR TITLE
Improve vagrant tests pt 1/x

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?

- Improves some parts of the E2E vagrant tests to run faster using snapshots
- Fixes use of Hiera paths and configs.
- Adds sample Hiera configs 

### What ticket does this PR close?

Connected to #139 
Connected to #89 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: examples/puppetmaster/vagrant/2_install_puppet_agent.sh [hunk 1]
```diff
@@ -7,6 +7,14 @@ set -eou pipefail
 
 source utils.sh
 
+snapshot_name="$(agent_snapshot_name $PUPPET_AGENT_VERSION)"
+
+echo "Restoring snapshot '$snapshot_name' with agent v$PUPPET_AGENT_VERSION (if available)..."
+if ! vagrant snapshot restore "$snapshot_name"; then
```
Comment (diverdane) [inline examples/puppetmaster/vagrant/2_install_puppet_agent.sh:13]: Oops, the `!` needs to be deleted. Should be:
```
if vagrant snapshot restore "$snapshot_name"; then
```

File: examples/puppetmaster/vagrant/utils.sh [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
 # COMPOSE_PROJECT_NAME environment variable be set appropriately.
 function puppet_host_port() {
-    echo "$(docker-compose port puppet 8140 | awk -F ':' '{print $2}')"
+  echo "$(docker-compose port puppet 8140 | awk -F ':' '{print $2}')"
+}
+
+# Return the name of a matching snapshot when provided with a puppet
+# agent version as a parameter.
+function agent_snapshot_name() {
+  agent_version="$1"
```
Comment (diverdane) [inline examples/puppetmaster/vagrant/utils.sh:24]: This line `agent_version=$1" isn't used. Maybe you intended to use it on the next line to keep param usage clear?

File: examples/puppetmaster/code/data/nodes/node01.us.cyber-ark.com.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+lookup_options:
+  '^conjur::host_factory_token':
+    convert_to: 'Sensitive'
+  '^conjur::authn_api_key':
+    convert_to: 'Sensitive'
+
+# conjur::account: 'cucumber'
```
Comment (diverdane) [inline examples/puppetmaster/code/data/nodes/node01.us.cyber-ark.com.yaml:8]: Nice, this sample config is handy!!!

File: examples/puppetmaster/vagrant/utils.sh (hunk 2 above)
Comment (sgnn7) [reply examples/puppetmaster/vagrant/utils.sh:24]: fixed

File: examples/puppetmaster/vagrant/2_install_puppet_agent.sh (hunk 1 above)
Comment (sgnn7) [reply examples/puppetmaster/vagrant/2_install_puppet_agent.sh]: fixed

**Pull request merged**